healthyregions/chicago-environment-explorer
# feat: add Public Housing overlay + dynamically read overlay data from Google Sheets

## Problem
We would like to add an overlay showing locations of Public Housing in Chicago (fixes #145, fixes #168)

We would also like an easier pattern for adding these overlays to the ChiVes frontend (fixes #173)

## Approach
* feat: create a new sheet for Overlays in our Variables spreadsheet, read/fetch these during the `yarn build:data` step to create `src/config/overlays.json`
    * Point data is read as CSV and parsed into GeoJSON format
    * Polygon data is read directly from GitHub (via URL at runtime) in GeoJSON format
* feat: at runtime read `overlays.json` and use this to render our list of possible overlays, show enabled overlays on the map
* fix: migrate all existing overlays to use this new pattern (order of overlays in the UI is determined by the order of the rows within the Google Sheet)

~~NOTE 1: currently only "Cooling Centers" + "Public Housing" (point data) are read using this pattern - more discovery is needed to determine the desired CSV format when `geometryType === 'polygon'`~~

~~NOTE 2: I still need to finish migrating the previously-established styles for "Cooling Centers" (black border around points, smaller point radius, etc)~~

NOTE 3: we still need to categorize / consolidate / choose colors for "Public Housing", so these points are currently not categorized by color. Tooltips for the new layer also need adjustment (currently reusing "Cooling Centers" tooltip, but the column names differ)

![Screenshot 2024-07-22 at 4 16 04 PM](https://github.com/user-attachments/assets/7d3fc39e-47c0-4fba-8a2e-782380f9608b)

## How to Test
1. Navigate to https://deploy-preview-179--chicago-env-explorer.netlify.app/map
    * You should see that the variable "Heat Index (Maximum)" is selected by default
    * You should see that the "Cooling Centers" overlay is enabled by default
2. Enable the new "Public Housing" overlay
    * You should see many more dots appear on the map - these are locations of Public Housing in Chicago
3. Experiment with other past overlays (Historical Redlining, Community Areas, Wards, Industrial or Non-Residential Areas, etc)
